config/fi_provider.m4

```diff
@@ -202,3 +203,53 @@ AC_DEFUN([_FI_PROVIDER_INVOKE],[
 		[])]
 	    )
 ])
+
+dnl
+dnl Internal; should not be called from provder .m4 scripts.
+dnl Helper macro that appends each providers build flags for pkg-config
+dnl
+dnl Arguments:
+dnl
+dnl $1: name of the provider
+dnl
+AC_DEFUN([_FI_PROVIDER_PC_ADD],[
+	 FI_PC_CFLAGS="$FI_PC_CFLAGS $$1_CFLAGS $$1_CPPFLAGS"
+	 FI_PC_PRIVATE_LIBS="$FI_PC_PRIVATE_LIBS $$1_LDFLAGS $$1_LIBS"
+])
+
+dnl
+dnl Substitute pkg-config variables after all providers have been initialized
+dnl If any package dependencies are ever needed, make use of
+dnl   FI_PC_REQUIRES_PKGS
+dnl   and
+dnl   FI_PC_PRIVATE_PKGS
+dnl
+AC_DEFUN([FI_PROVIDER_SETUP_PC],[
+	FI_PC_PRIVATE_LIBS="$FI_PC_PRIVATE_LIBS $LIBS"
+
+	dnl Try and avoid duplication of private library flags
+	dnl Inefficient but easier than tracking within each provider macro
+	seen=""
+	for arg in $FI_PC_PRIVATE_LIBS
+	do
+	  found=no
+	  for s in $seen
+	  do
+	    AS_IF([test "$s" = "$arg"], [found=yes])
+	  done
+	  AS_IF([test "$found" = "no"], [seen="$seen $arg"])
+	done
+	FI_PC_PRIVATE_LIBS=$seen
+
+	AS_IF([test "x$enable_shared" != xyes],
+	 [FI_PC_PUBLIC_LIBS="$FI_PC_PUBLIC_LIBS $FI_PC_PRIVATE_LIBS"
+	 FI_PC_REQUIRES_PKGS="$FI_PC_REQUIRES_PKGS $FI_PC_PRIVATE_PKGS"])
```

Comment:
No, that is just another convenience for listing the private required packages on the public required packages line if a static-only build is configured.  As it stands now, the libfabric .pc doesn't list `pkg-config` package dependencies.  This could change if, for example, libfabric used the result of `FI_PKG_CHECK_MODULE` to add a required package dependency via `FI_PC_PRIVATE_PKGS` if necessary.  Right now I think only the `gni` provider uses the pkg macro though.